src/java_binding/java/java-binding/src/main/java/com/risingwave/java/binding/Iterator.java

```diff
@@ -1,11 +1,35 @@
 package com.risingwave.java.binding;
 
+import com.risingwave.java.binding.rpc.MetaClient;
+import com.risingwave.proto.Catalog.Table;
+import com.risingwave.proto.Hummock.HummockVersion;
+import com.risingwave.proto.JavaBinding.KeyRange;
+import com.risingwave.proto.JavaBinding.ReadPlan;
+
 public class Iterator implements AutoCloseable {
-    final long pointer;
-    boolean isClosed;
+    private final long pointer;
+    private boolean isClosed;
+
+    public Iterator(
+            MetaClient metaClient,
```

Comment:
In practice, the iterator does not necessarily need to connect to the meta node to get the meta data. 

For example, in Spark, a distributed compute engine, only its driver (a central controller) needs to connect to meta node to get the necessary meta data, and pin the version, and then it will generate an execution plan and send these meta data to the executors (a set of workers that really do the work). The executors will create the iterator with these meta data and do the scan, during which there is no need to connect to the meta node. All it needs for execution are the object store url and a read plan. So we'd better move the logic of getting meta data outside the iterator.